website/docs/r/lightsail_container_service.html.markdown

```diff
@@ -0,0 +1,213 @@
+---
+subcategory: "Lightsail"
+layout: "aws"
+page_title: "AWS: aws_lightsail_container_service"
+description: |- 
+  Provides a resource to manage Lightsail container service
+---
+
+# Resource: aws_lightsail_container_service
+
+An Amazon Lightsail container service is a highly scalable compute and networking resource on which you can deploy, run,
+and manage containers. For more information, see
+[Container services in Amazon Lightsail](https://lightsail.aws.amazon.com/ls/docs/en_us/articles/amazon-lightsail-container-services).
+
+**Note:** For more information about the AWS Regions in which you can create Amazon Lightsail container services,
+see ["Regions and Availability Zones in Amazon Lightsail"](https://lightsail.aws.amazon.com/ls/docs/overview/article/understanding-regions-and-availability-zones-in-amazon-lightsail).
+
+## Example Usage
+
+```terraform
+# create a new Lightsail container service
+resource "aws_lightsail_container_service" "my_container_service" {
+  name = "container-service-1"
+  power = "nano"
+  scale = 1
+  is_disabled = false
+
+  # deployment {
+    # example below
+  # }
+
+  # public_domain_names {
+    # example below
+  # }
+
+  tags = {
+    foo1 = "bar1"
+    foo2 = ""
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name for the container service. Names must be of length 1 to 63, and be
+  unique within each AWS Region in your Lightsail account.
+* `power` - (Required) The power specification for the container service. The power specifies the amount of memory,
+  the number of vCPUs, and the monthly price of each node of the container service.
+* `scale` - (Required) The scale specification for the container service. The scale specifies the allocated compute
+  nodes of the container service.
+* `is_disabled` - (Optional) A Boolean value indicating whether the container service is disabled. Defaults to `false`.
+* `deployment` - (Optional) A deployment specifies the containers that will be launched on the container service and
+  their settings, such as the ports to open, the environment variables to apply, and the launch command to run. It also
+  specifies the container that will serve as the public endpoint of the deployment and its settings, such as the HTTP or
+  HTTPS port to use, and the health check configuration. Defined below.
+* `public_domain_names` - (Optional) The public domain names to use with the container service, such as example.com
+  and www.example.com. You can specify up to four public domain names for a container service. The domain names that you
+  specify are used when you create a deployment with a container configured as the public endpoint of your container
+  service. If you don't specify public domain names, then you can use the default domain of the container service.
+  Defined below.
+    * WARNING: You must create and validate an SSL/TLS certificate before you can use public domain names with your
```

Comment:
Can a certificate be created via Terraform for this to help the user do everything through the HCL interface?

Reply:
When I was doing this pull request as an intern at Lightsail, it was not possible then. Perhaps @mlburggr has newer information.

Reply:
@AdamTylerLynch  and @KaguraKagura  I currently have an implementation of the  `certificate` resource in my separate lightsail provider and plan on opening up a PR to this provider with the resource as soon as my other PR is merged #18663 as this PR has internal waiters written that I would like to use when writing other lightsail resources. The example of the `certificate` resource is currently here: https://github.com/DeYoungTech/terraform-provider-awslightsail/blob/main/internal/lightsail/certificate.go